openapi.yaml

```diff
@@ -16,175 +16,301 @@ host: api.oregonstate.edu
           description: Successful response
           schema:
-            $ref: '#/definitions/PetResult'
+            $ref: '#/definitions/VersionHistoryResult'
         404:
-          description: PetResource not found
+          description: Note not found
           schema:
             $ref: '#/definitions/Errors'
         500:
-          description: Internal Server Error
+          description: Internal server error
           schema:
             $ref: '#/definitions/Errors'
-parameters:
-  authorization:
-    name: Authorization
-    in: header
-    type: string
-    required: true
-    description: '"Bearer [token]" where [token] is your OAuth2 access token'
-  pageNumber:
-    name: page[number]
-    in: query
-    type: integer
-    required: false
-    description: Page number of results
-    default: 1
-    minimum: 1
-  pageSize:
-    name: page[size]
-    in: query
-    type: integer
-    required: false
-    description: Number of results to return
-    default: 25
-    maximum: 500
-    minimum: 1
 definitions:
-  Meta:
+  noteID:
+    type: string
+    description: A unique ID of a note concatenating the student ID that the note is for with a random string
+    example: 123456789-abc123
+  note:
+    type: string
+    description: The contents of the note
+  studentID:
+    type: string
+    description: The 9-digit OSU ID of the student who the note is on
+    example: '123456789'
+  creatorID:
+    type: string
+    description: The 9-digit OSU ID of the advisor who created the note
+    example: '987654321'
+  source:
+    type: string
+    enum: *SOURCES
+    description: The data source of the note
+  permissions:
+    type: string
+    enum:
+      - advisor
+      - advisors
+      - student
+    description: "A string that defines the permissions of who can view the note. 'advisor' indicates that only
+      the advisor who created the note may view it. 'advisors' indicates that all advisors can view the note.
+      'student' indicates that all advisors as well as the student who the note is on may view the note."
+  context:
+    type: object
     properties:
-      totalResults:
-        type: integer
-        description: Total number of results
-        example: 10
-      totalPage:
-        type: integer
-        description: Total number of pagess
-        example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
+      contextType:
+        type: string
+        description: The context type of the note
+        example: 'course'
+      contextID:
+        type: string
+        description: The context id of the note
+        example: 'CS-101'
+  dateCreated:
+    type: string
+    format: date-time
+    description: The original date that the note was created
+    example: 2017-07-21T17:32:28Z
+  dateModified:
```

Comment:
From our discussion, I think the field should be called `lastModified`? But I'm fine in either way, just make sure our openapi are consistent with the API doc.